### PR TITLE
[CAS-301] Replace Glide with Coil

### PR DESCRIPTION
**Pull request by adasiewiczr**
### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android/build.gradle [hunk 1]
```diff
@@ -102,12 +102,6 @@ dependencies {
 
     implementation Dependencies.gson
 
-    implementation Dependencies.glideRedirect
-    implementation(Dependencies.glide) {
-        exclude group: "com.android.support"
-    }
-    kapt Dependencies.glideCompiler
```
Comment (zsmb13) [inline stream-chat-android/build.gradle:109]: This means we can also disable `kapt` in this module

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/ImageLoader.kt [hunk 2]
```diff
@@ -41,13 +43,28 @@ public object ImageLoader {
...
+
+    @JvmStatic
+    @JvmOverloads
+    public fun ImageView.load(
         uri: String?,
-        @DrawableRes placeholderResId: Int?,
+        @DrawableRes placeholderResId: Int? = null,
+        onStart: () -> Unit = {},
+        onComplete: () -> Unit = {},
+    ): Unit = loadAny(uri, placeholderResId, onStart, onComplete)
```
Comment (zsmb13) [inline stream-chat-android/src/main/java/com/getstream/sdk/chat/ImageLoader.kt:59]: Looks like we should be able to merge these two methods into just one

Comment (adasiewiczr) [reply stream-chat-android/src/main/java/com/getstream/sdk/chat/ImageLoader.kt:59]: I didn't want to expose `loadAny` outside this class. IMO it is better to use type-safe versions

Comment (zsmb13) [reply stream-chat-android/src/main/java/com/getstream/sdk/chat/ImageLoader.kt:59]: Nevermind, didn't realize one of these took a `Uri` and the other a `String` as its parameter 🤦

**Review by zsmb13 — APPROVED** [review]

**Review by JcMinarro — COMMENTED** [review]
Great!!!
We should add a comment into our Changelog that our SDK doesn't depend anymore on Glide :tada:

**Comment by tschellenbach** [comment]
This is great 👍

**Review by bychkovdmitry — DISMISSED** [review]

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**